### PR TITLE
Separate frontend from backend

### PR DESCRIPTION
**Pull request by lucycy98**
refactored app so that the frontend and backend is separate from each other.

### CODE REVIEW

**Comment by preetpatel** [comment]
sorry this is taking me a bit.. havent gotten the time to do anything yet

**Comment by gargi-bansal** [comment]
If you're separating the backend from the front end so that it has its own config/yarn.lock files/etc, shouldn't the front-end also have its own yarn.lock/package.json? @PreetPatel correct me if I'm wrong?

**Comment by preetpatel** [comment]
> If you're separating the backend from the front end so that it has its own config/yarn.lock files/etc, shouldn't the front-end also have its own yarn.lock/package.json? @PreetPatel correct me if I'm wrong?

yup you are right gargi.. Working on a better solution rn

**Comment by preetpatel** [comment]
Ok so after doing some looking.. it is not possible to easily change the src/ folder name to frontend (Which would be the only change required to satisfy this problem). 
I'm content in having the following distinction:
src/ - contains the frontend code
server/ contains the backend code

There is no point in further separating the folders as it will require us to have 2 different node module folders, package.json and scripts adding additional load to you, the developer as you will be downloading 2x the number of duplicated modules.

The reason why I initially structured the code like this is because this is the new recommended way since frontend files are served by the same endpoint that acts as a server on a different path on the same hostname. For the purposes of this project, this is a perfectly good structure as it keeps all development on one repo.
TLDR; Having a mono-repo with 2 different folders with their own package.json and .gitignore and node_modules is the same as having 2 different repos so what real benefit are you gaining? 

I suggest closing this PR
